### PR TITLE
Merge feature/sb17 into origin/main

### PR DESCRIPTION
**Pull request by niranjanyardi**
b/360216121

Removes windows based builds and SB15, SB16 code
---------

Co-authored-by: Madhura Jayaraman <madjay@google.com>
Co-authored-by: Hao <131711973+haozheng-cobalt@users.noreply.github.com>
Co-authored-by: Jelle Foks <jfoks@google.com>
Co-authored-by: Oscar Vestlie <oxv@google.com>
Co-authored-by: Yijia Zhang <45114178+yjzhang111@users.noreply.github.com>
Co-authored-by: Kaido Kert <kaidokert@google.com>
Co-authored-by: Colin Liang <zhongqi.liang.4u@gmail.com>
Co-authored-by: Colin Liang <colinliang@google.com>
Co-authored-by: xiaomings <xiaomings@google.com>

### CODE REVIEW

File: starboard/file.h [hunk 1]
```diff
@@ -129,38 +129,7 @@ static inline bool SbFileIsValid(SbFile file) {
...
-// DEPRECATED with SB_API_VERSION 16
-//
-// Closes |file|. The return value indicates whether the file was closed
-// successfully.
-//
-// |file|: The absolute path of the file to be closed.
-SB_EXPORT bool SbFileClose(SbFile file);
-
-#endif  // SB_API_VERSION < 17
+#endif
```
Comment (madhurajayaraman) [inline starboard/file.h:132]: This conditional needs to be removed in a follow up CL. I can do it with some more starboard cleanup once this goes in.

File: starboard/nplb/player_test_util.cc [hunk 2]
```diff
@@ -60,7 +60,6 @@ const char* kAudioOnlyTestFiles[] = {
     "iamf_base_profile_stereo_ambisonics.dmp",
     "iamf_simple_profile_5_1.dmp",
 #endif  // SB_API_VERSION >= 15
-    "sintel_5s_pcm_s16le.dmp",
```
Comment (madhurajayaraman) [inline starboard/nplb/player_test_util.cc:63]: Where is this deletion from?

**Comment by madhurajayaraman** [comment]
From a first pass, everything looks good. Can you run on_device tests? Also someone with more permissions than I have should be able to remove the SB15 and 16 build requirements for the PR.

File: starboard/file.h (hunk 1 above)
Comment (niranjanyardi) [reply starboard/file.h:132]: ack

File: cobalt/h5vcc/h5vcc_settings.cc [hunk 3]
```diff
@@ -219,13 +219,6 @@ std::string H5vccSettings::GetPersistentSettingAsString(
     absl::optional<std::string> json = base::WriteJson(value);
     return json.value_or(std::string());
   }
-
-  if (key.compare(network::kProtocolFilterKey) == 0) {
```
[the remark this answers is not in the document]
Comment (niranjanyardi) [reply cobalt/h5vcc/h5vcc_settings.cc]: Fixed. This seems to be present in main but not in sb17 - the git merge got rid of the code in main

File: starboard/nplb/player_test_util.cc (hunk 2 above)
Comment (niranjanyardi) [reply starboard/nplb/player_test_util.cc]: same as the previous resolution

**Comment by niranjanyardi** [comment]
> From a first pass, everything looks good. Can you run on_device tests? Also someone with more permissions than I have should be able to remove the SB15 and 16 build requirements for the PR.

Added on_device. Oscar has disabled the sb15,16 builds

**Comment by niranjanyardi** [comment]
Android failures seem unrelated: https://paste.googleplex.com/5596628095598592

**Review by kaidokert — APPROVED** [review]

**Pull request merged**